src/components/EdgeSlicing.vue

```
@@ -0,0 +1,446 @@
+      }
+    }
+
+    function formatDate(date: Date) {
+      const dateString = formatShortDate(date);
+      const [month, day, year] = dateString.split('/');
+      return `${year}-${month}-${day}`;
+    }
+
+    // Assume start var + end are the same
+    watch([startEdgeVar], () => {
+      endEdgeVar.value = startEdgeVar.value;
+    });
+
+    const cleanedEdgeVariables = computed(() => Object.keys(store.state.edgeAttributes).filter((varName) => !isInternalField(varName)));
+
+    // Compute the min and max times for numbers or date
+    const varRange = computed(() => {
+      const range: Date[] | number[] | string[] = [];
+      if (startEdgeVar.value !== null && endEdgeVar.value !== null && originalNetwork.value !== null) {
+        // Loop through all edges, return min and max time values
+        originalNetwork.value.edges.forEach((edge: Edge, i: number) => {
+          // Check for dates
+          let startVar: number = parseFloat(`${edge[startEdgeVar.value]}`);
+          let endVar: number = parseFloat(`${edge[endEdgeVar.value]}`);
+          if (isDate.value) {
+            startVar = Date.parse(`${edge[startEdgeVar.value]}`);
+            endVar = Date.parse(`${edge[endEdgeVar.value]}`);
+          }
+          if (i === 0) {
+            range[0] = startVar;
+            range[1] = endVar;
+          }
+          if (startVar < range[0]) {
+            range[0] = startVar;
+          }
+          if (endVar > range[1]) {
+            range[1] = endVar;
+          }
+        });
+      }
+      // Format date
+      if (isDate) {
+        dateFormatted.value = [formatDate(new Date(range[0])), formatDate(new Date(range[1]))];
+      }
+      return range;
+    });
+
+    watch([dateFormatted], () => {
+      if (isDate.value) {
+        varRange.value[0] = new Date(dateFormatted.value[0]);
+        varRange.value[1] = new Date(dateFormatted.value[1]);
+      }
+    });
+
+    function sliceNetwork() {
```

This fails silently if the range specified by user is outside of the values of the data. For example, I have data from 0 to 31 and if I specify -100 to 31, it fails without alerting the user. We could have some simple validation and show errors to the user when it happens

---

resolved for both number inputs _and_ date inputs